src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Documents/EditorDocumentManagerListener.cs

```diff
@@ -132,20 +136,19 @@ internal async void ProjectManager_Changed(object sender, ProjectChangeEventArgs
             }
         }
 
-#pragma warning disable VSTHRD100 // Avoid async void methods
-        private async void Document_ChangedOnDisk(object sender, EventArgs e)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+        private void Document_ChangedOnDisk(object sender, EventArgs e)
         {
             try
             {
-                // This event is called by the EditorDocumentManager, which runs on the UI thread.
-                // However, due to accessing the project snapshot manager, we need to switch to
-                // running on the project snapshot manager's specialized thread.
-                await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(() =>
-                {
-                    var document = (EditorDocument)sender;
-                    _projectManager.DocumentChanged(document.ProjectFilePath, document.DocumentFilePath, document.TextLoader);
-                }, CancellationToken.None).ConfigureAwait(false);
+                _joinableTaskContext.Factory.Run(async () =>
```

Comment:
This lambda should have braces around it IMO. It's hard to see what the `ConfigureAwait(false)` is on